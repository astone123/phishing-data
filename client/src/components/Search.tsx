import * as React from 'react';

class Search extends React.Component<IAppState & Notify, { query: string }> {
  constructor(props: IAppState & Notify) {
    super(props);
    this.state = { query: props.query || '' };

    this.handleQueryChange = this.handleQueryChange.bind(this);
    this.onChange = this.onChange.bind(this);
  }

  handleQueryChange(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();

    this.props.notify({ query: this.state.query });
  }

  onChange(e: React.ChangeEvent<HTMLInputElement>) {
    const target = e.target as HTMLInputElement;
    this.setState({ query: target.value });
  }

  render() {
    return (
      <form onSubmit={this.handleQueryChange}>
        <input
          autoFocus={true}
          type="search"
          placeholder="Filter by IP, Target, or URL"
          value={this.state.query || ''}
          onChange={this.onChange}
        />
      </form>
    );
  }
}

export default Search;
