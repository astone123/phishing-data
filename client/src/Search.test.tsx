import * as React from 'react';
import * as renderer from 'react-test-renderer';
import Search from './components/Search';

test('Test Bar Chart Component', () => {
  const tree = renderer.create(<Search query="" />).toJSON();
  expect(tree).toMatchSnapshot();
});
