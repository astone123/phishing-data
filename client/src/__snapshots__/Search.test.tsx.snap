// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Bar Chart Component 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    autoFocus={true}
    onChange={[Function]}
    placeholder="Filter by IP, Target, or URL"
    type="search"
    value=""
  />
</form>
`;
